src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -267,16 +267,31 @@ export class BrowserRenderer {
       this.trySetSelectValueFromOptionElement(newDomElementRaw);
     } else if (deferredValuePropname in newDomElementRaw) {
       // Situation 2
-      const deferredValue: string | null = newDomElementRaw[deferredValuePropname];
-      setDeferredElementValue(newDomElementRaw, deferredValue);
+      setDeferredElementValue(newDomElementRaw, newDomElementRaw[deferredValuePropname]);
     }
   }
 
   private trySetSelectValueFromOptionElement(optionElement: HTMLOptionElement) {
     const selectElem = this.findClosestAncestorSelectElement(optionElement);
-    if (selectElem && (deferredValuePropname in selectElem) && selectElem[deferredValuePropname] === optionElement.value) {
-      setDeferredElementValue(selectElem, optionElement.value);
-      delete selectElem[deferredValuePropname];
+    if (selectElem && (deferredValuePropname in selectElem)) {
+      if (selectElem.type === 'multiple-select') {
+        return this.trySetMultipleSelectValueFromOptionElement(selectElem, optionElement);
+      } else {
+        return this.trySetSingleSelectValueFromOptionElement(selectElem, optionElement);
+      }
+    }
+    return false;
+  }
+
+  private trySetMultipleSelectValueFromOptionElement(selectElement: HTMLSelectElement, optionElement: HTMLOptionElement) {
+    optionElement.selected = selectElement[deferredValuePropname].indexOf(optionElement.value) !== -1;
+    return true;
```

Comment:
I can't think of any drawback to not deleting the `deferredValuePropname` property here (and I understand why you can't) but if you think there are any risks please say so.